lib/security/sha256.js

```diff
@@ -0,0 +1,21 @@
+var CryptoJS = require("crypto-js");
+var q = require('q');
+
+/**
+* Generate a SHA256 hash value for some given text.
+* @param text - the text to generate a hash value for,
+*/
+function hash(text) {
+  var deferred = q.defer();
```

Comment:
Same as above, it doesn't look like an asynchronous operation, are the promises required?